src/client/options.rs

```diff
@@ -1285,6 +1297,18 @@ impl ClientOptions {
                 };
             }
         }
+
+        let max_connecting = if let Some(mc) = conn_str.max_connecting {
+            // maxConnecting must be greater than 0
+            if mc == 0 {
+                None
+            } else {
+                Some(mc)
+            }
+        } else {
+            None
+        };
```

Comment:
This method should just copy the fields from the connection string directly to the client options regardless of their validity. I think we should move this logic to the `ClientOptions::validate` method instead, and return an error if the value for `max_connecting` is invalid.

Reply:
That makes sense!